### PR TITLE
parpack --- add support for trilinos mpi blockvector

### PR DESCRIPTION
**Pull request by asartori86**

### CODE REVIEW

File: include/deal.II/lac/parpack_solver.h [hunk 1]
```diff
@@ -234,6 +234,12 @@ class PArpackSolver : public Subscriptor
   void reinit(const dealii::IndexSet &locally_owned_dofs );
 
   /**
+   * Initialize internal variables when working with BlockVectors
+   */
+  void reinit(const dealii::IndexSet &locally_owned_dofs,
```
Comment (davydden) [inline include/deal.II/lac/parpack_solver.h]: don't need `dealii::` namespace here.

File: include/deal.II/lac/parpack_solver.h [hunk 2]
```diff
@@ -234,6 +234,12 @@ class PArpackSolver : public Subscriptor
   void reinit(const dealii::IndexSet &locally_owned_dofs );
 
   /**
+   * Initialize internal variables when working with BlockVectors
```
Comment (davydden) [inline include/deal.II/lac/parpack_solver.h]: please add some description on `@p locally_owned_dofs` and `@p block`

File: include/deal.II/lac/parpack_solver.h [hunk 3]
```diff
@@ -516,6 +522,54 @@ void PArpackSolver<VectorType>::reinit(const dealii::IndexSet &locally_owned_dof
...
+  lworkev = additional_data.symmetric ?
+            0 /*not used in symmetric case*/
+            :
+            3*ncv;
+  workev.resize (lworkev, 0.);
+
+  select.resize (ncv, 0);
+
+  // deal.II vectors:
+  src.reinit (block,mpi_communicator);
```
Comment (davydden) [inline include/deal.II/lac/parpack_solver.h]: everything but this line is the same for the two functions. I would suggest to refactor the code (maybe add a function in` private:` which is has the common part and then the two functions would call this private one.

**Comment by davydden** [comment]
an entry to `changes.h` would be good to have (btw, the support is for block vectors in general, not only for trilinos).

**Comment by asartori86** [comment]
@davydden, thanks for your feedbacks. I've addressed your comments, at least I hope :D 

May I ask a question related to Arpack/PArpack? I was computing the eigenvalues of the laplacian in 2d on a square with side of lenght \pi. In order to get the **smallest** eigenvalues I had to choose **largest_magnitude**. Is this expected?

**Comment by davydden** [comment]
thanks, look good to me :+1:

**Comment by davydden** [comment]
as for your question, since we do shift-and-invert spectral transformation, then, say for zero shift and positive eignevalues, the largest eigenvalue of the transformed problem will correspond to the smallest eigenvalue of the original problem

**Comment by asartori86** [comment]
thanks @davydden for your explanation!

**Pull request merged**